projects/NeRF-Det/nerfdet/data_preprocessor.py

```diff
@@ -0,0 +1,583 @@
+            for aug_id in range(num_augs):
+                single_aug_batch_data = self.simple_process(
+                    data[aug_id], training)
+                aug_batch_data.append(single_aug_batch_data)
+            return aug_batch_data
+
+        else:
+            return self.simple_process(data, training)
+
+    def simple_process(self, data: dict, training: bool = False) -> dict:
+        """Perform normalization, padding and bgr2rgb conversion for img data
+        based on ``BaseDataPreprocessor``, and voxelize point cloud if `voxel`
+        is set to be True.
+
+        Args:
+            data (dict): Data sampled from dataloader.
+            training (bool): Whether to enable training time augmentation.
+                Defaults to False.
+
+        Returns:
+            dict: Data in the same format as the model input.
+        """
+        if 'img' in data['inputs']:
+            batch_pad_shape = self._get_pad_shape(data)
+
+        data = self.collate_data(data)
+        inputs, data_samples = data['inputs'], data['data_samples']
+        batch_inputs = dict()
+
+        if 'points' in inputs:
+            batch_inputs['points'] = inputs['points']
+
+            if self.voxel:
+                voxel_dict = self.voxelize(inputs['points'], data_samples)
+                batch_inputs['voxels'] = voxel_dict
+
+        if 'imgs' in inputs:
+            imgs = inputs['imgs']
+
+            if data_samples is not None:
+                # NOTE the batched image size information may be useful, e.g.
+                # in DETR, this is needed for the construction of masks, which
+                # is then used for the transformer_head.
+                batch_input_shape = tuple(imgs[0].size()[-2:])
+                for data_sample, pad_shape in zip(data_samples,
+                                                  batch_pad_shape):
+                    data_sample.set_metainfo({
+                        'batch_input_shape': batch_input_shape,
+                        'pad_shape': pad_shape
+                    })
+
+                if self.boxtype2tensor:
+                    samplelist_boxtype2tensor(data_samples)
+                if self.pad_mask:
+                    self.pad_gt_masks(data_samples)
+                if self.pad_seg:
+                    self.pad_gt_sem_seg(data_samples)
+
+            if training and self.batch_augments is not None:
+                for batch_aug in self.batch_augments:
+                    imgs, data_samples = batch_aug(imgs, data_samples)
+            batch_inputs['imgs'] = imgs
+        # Hard code here, will be changed later.
+        # if len(inputs['depth']) != 0:
+        if 'depth' in inputs.keys():
+            batch_inputs['depth'] = inputs['depth']
+        batch_inputs['lightpos'] = inputs['lightpos']
+        batch_inputs['nerf_sizes'] = inputs['nerf_sizes']
+        batch_inputs['denorm_images'] = inputs['denorm_images']
+        batch_inputs['raydirs'] = inputs['raydirs']
```

Comment:
some of the parameters can be put into the meta_info?

Reply:
These parameters play a similar role to 'img' earlier, so I put them here.